### PR TITLE
Access Working

### PR DESCRIPTION
**Pull request by JashonMenya**

### CODE REVIEW

File: itineraries/views.py [hunk 1]
```diff
@@ -32,16 +34,48 @@ def hello_world(request):
...
+
+'''
+Registers a new user and redirects to the home page.
+
+When a new user submits registration information, it's saved to the database, and then the user is redirected to the home page.
+'''
+
+def signup(request):
     if request.method == 'POST':
-        form = UserRegistrationForm(request.POST)
```
Comment (toddvharper) [inline itineraries/views.py:37]: This should stay UserRegistrationForm.

File: itineraries/views.py [hunk 2]
```diff
@@ -1,8 +1,10 @@
 from django.shortcuts import redirect, render
-from django.contrib.auth import login
+from django.contrib.auth import authenticate, login, logout
 from django.contrib.auth.decorators import login_required
-from .forms import UserRegistrationForm, PreferencesForm
```
Comment (toddvharper) [inline itineraries/views.py:4]: We should import UserRegistrationForm.

File: itineraries/views.py [hunk 3]
```diff
@@ -32,16 +34,48 @@ def hello_world(request):
...
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            login(request, user)
+            return redirect('home')
+        else:
+            return redirect('home')
+    return render(request, 'header-main.html')
```
Comment (toddvharper) [inline itineraries/views.py]: I think instead of rendering header-main, we should render index.html.

File: itineraries/views.py [hunk 4]
```diff
@@ -32,16 +34,48 @@ def hello_world(request):
...
-            user = form.save()
-            login(request, user)
-            return redirect('../hello')
+            form.save()
+            return redirect('home')
     else:
-        form = UserRegistrationForm()
-    return render(request, 'auth/signup.html', {'form': form})
+        form = UserCreationForm()
+    return render(request, 'header-main.html', {'form': form})
```
Comment (toddvharper) [inline itineraries/views.py]: I think instead of rendering header-main, we should render index.html.

**Review by toddvharper — APPROVED** [review]
Looks good! A couple of things to add/change in the next iteration:
1) Unless we add an email or OTP verification workflow, I think it would make sense to automatically log in the user right when they register, rather than making them register and then sign in.
2) When I enter invalid information in the registration form, instead of showing me what's wrong (ie. username already taken), I get an error page saying that 'home' is not a valid template. I can't look into it right now, but I suspect that Django expects an actual path to an HTML template in the render field instead of a named reference.

But really great work, JJ! I'm excited for this development :)

**Review by SamRichter01 — APPROVED** [review]

**Pull request merged**